src/prefect/orion/utilities/schemas.py

```diff
@@ -126,13 +126,16 @@ def orjson_dumps(v: Any, *, default: Any) -> str:
 
 def orjson_dumps_non_str_keys(v: Any, *, default: Any) -> str:
```

Comment:
We'll probably want to rename this as like... `orjson_dumps_extra_compatible` or something :D (maybe there's a better name than that)

Reply:
good catch, thanks! I also added a test for compatibility with numpy types but this introduces a new requirement in `requirements-dev.txt`, so let me know if you want to maintain it